TypeDB.java

```diff
@@ -34,42 +35,26 @@
     public static final String DEFAULT_ADDRESS = "localhost:1729";
 
     public static TypeDBClient coreClient(String address) {
-        return CoreClient.create(address);
+        return CoreClient.create(address, TypeDBOptions.core());
     }
 
-    public static TypeDBClient coreClient(String address, int parallelisation) {
-        return CoreClient.create(address, parallelisation);
-    }
-
-    public static TypeDBClient.Cluster clusterClient(String address, boolean tlsEnabled) {
-        return ClusterClient.create(set(address), tlsEnabled, null);
-    }
-
-    public static TypeDBClient.Cluster clusterClient(String address, boolean tlsEnabled, String tlsRootCA) {
-        return ClusterClient.create(set(address), tlsEnabled, tlsRootCA);
+    public static TypeDBClient coreClient(String address, TypeDBOptions options) {
+        return CoreClient.create(address, options);
     }
 
-    public static TypeDBClient.Cluster clusterClient(String address, boolean tlsEnabled, int parallelisation) {
-        return ClusterClient.create(set(address), tlsEnabled, null, parallelisation);
-    }
-
-    public static TypeDBClient.Cluster clusterClient(String address, boolean tlsEnabled, String tlsRootCA, int parallelisation) {
-        return ClusterClient.create(set(address), tlsEnabled, tlsRootCA, parallelisation);
-    }
-
-    public static TypeDBClient.Cluster clusterClient(Set<String> addresses, boolean tlsEnabled) {
-        return ClusterClient.create(addresses, tlsEnabled, null);
+    public static TypeDBClient coreClient(String address, int parallelisation) {
+        return CoreClient.create(address, TypeDBOptions.core(), parallelisation);
     }
 
-    public static TypeDBClient.Cluster clusterClient(Set<String> addresses, boolean tlsEnabled, String tlsRootCA) {
-        return ClusterClient.create(addresses, tlsEnabled, tlsRootCA);
+    public static TypeDBClient coreClient(String address, TypeDBOptions options, int parallelisation) {
+        return CoreClient.create(address, options, parallelisation);
     }
 
-    public static TypeDBClient.Cluster clusterClient(Set<String> addresses, boolean tlsEnabled, int parallelisation) {
-        return ClusterClient.create(addresses, tlsEnabled, null, parallelisation);
+    public static TypeDBClient.Cluster clusterClient(String address, TypeDBOptions.Cluster options) {
```

Comment:
All variants of `TypeDB::clusterClient` requires the user to pass `TypeDBOptions` object. This is because the TLS and username/password are mandatory parameters.